driver-core/src/main/java/com/datastax/driver/core/DirectedGraph.java

```diff
@@ -65,16 +67,21 @@ void addEdge(V from, V to) {
 
         Queue<V> queue = new LinkedList<V>();
 
-        for (Map.Entry<V, Integer> entry : vertices.entrySet()) {
-            if (entry.getValue() == 0)
-                queue.add(entry.getKey());
+        // Sort vertices so order of evaluation is always the same (instead of depending on undefined map order behavior)
+        List<V> orderedVertices = new ArrayList<V>(vertices.keySet());
+        Collections.sort(orderedVertices, comparator);
```

Comment:
Not too concerned about performance as number of UDTs should be rather small, and we only sort at top level and within adjacency lists.